src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -629,13 +617,11 @@ private void deleteFeedbackQuestionCascade(
         // Cascade delete responses for question.
         frLogic.deleteFeedbackResponsesForQuestionAndCascade(questionToDelete.getId(), true);
 
-        List<FeedbackQuestionAttributes> questionsToShiftQnNumber = null;
-        try {
-            questionsToShiftQnNumber = getFeedbackQuestionsForSession(feedbackSessionName, courseId);
-        } catch (EntityDoesNotExistException e) {
+        List<FeedbackQuestionAttributes> questionsToShiftQnNumber =
+                getFeedbackQuestionsForSession(feedbackSessionName, courseId);
+        if (fsLogic.getFeedbackSession(feedbackSessionName, courseId) == null) {
```

Comment:
The check should do before `getFeedbackQuestionsForSession` ? Same for other place?

Reply:
Yes. Moved them before questions for session retrieval